test/gtest/interpolate.hpp

```diff
@@ -0,0 +1,378 @@
+                                                           input_dev.get(),
+                                                           output.desc,
+                                                           output_dev.get(),
+                                                           scale_factors.desc,
+                                                           scale_factors_dev.get(),
+                                                           mode,
+                                                           align_corners);
+        }
+        fflush(stdout);
+        ASSERT_EQ(status, miopenStatusSuccess);
+
+        output.data = handle.Read<T>(output_dev, output.data.size());
+    }
+
+    void Verify()
+    {
+        double threshold = std::numeric_limits<T>::epsilon();
+
+        auto error = miopen::rms_range(ref_output, output);
+
+        ASSERT_EQ(miopen::range_distance(ref_output), miopen::range_distance(output));
+        EXPECT_LT(error, threshold * 10) << "Error output beyond tolerance Error:" << error
+                                         << ",  Thresholdx10: " << threshold * 10;
+    }
+    InterpolateTestCase interpolate_config;
+
+    tensor<T> input;
+    tensor<T> output;
+    tensor<T> ref_output;
+    tensor<float> scale_factors;
+
+    miopenInterpolateMode_t mode;
+    bool align_corners;
+
+    miopen::Allocator::ManageDataPtr input_dev;
+    miopen::Allocator::ManageDataPtr output_dev;
+    miopen::Allocator::ManageDataPtr scale_factors_dev;
+};
+
+// BACKWARD TEST
+template <typename T = float>
+struct InterpolateTestBwd : public ::testing::TestWithParam<InterpolateTestCase>
+{
+protected:
+    void SetUp() override
+    {
+        auto&& handle      = get_handle();
+        interpolate_config = GetParam();
+
+        auto in_dim      = interpolate_config.GetInput();
+        auto in_grad_dim = in_dim;
+        auto size        = interpolate_config.size;
+        mode             = interpolate_config.mode;
+        align_corners    = interpolate_config.align_corners;
+
+        if(mode != MIOPEN_INTERPOLATE_MODE_NEAREST)
+        {
+            scale_factors = tensor<float>{size.size()};
+            for(int i = 0; i < size.size(); i++)
+                scale_factors[i] = interpolate_config.scale_factors[i];
+        }
+        else
+        {
+            scale_factors = tensor<float>{3};
+            for(int i = 0; i < size.size(); i++)
+                scale_factors[i] = interpolate_config.scale_factors[i];
+            for(int i = size.size(); i < 3; i++)
+                scale_factors[i] = 0;
+        }
+
+        auto out_grad_dim = std::vector<size_t>({in_dim[0], in_dim[1]});
+        for(int i = 0; i < size.size(); i++)
+        {
+            if(scale_factors[i] != 0)
+                out_grad_dim.push_back(ceil(static_cast<size_t>(in_dim[i + 2] * scale_factors[i])));
+            else
+                out_grad_dim.push_back(size[i]);
```

Comment:
you forgot to set scale_factor, which then lead to CPU and GPU both use scale_factor = 0 in this case 😒😒

Reply:
scale_factor = 0 is not an invalid case, scale_factor is always recompute by kernel compute_linear_scale_factor

Reply:
I pointed out that because in TestFwd you set scale_factor in this case. If there are not any problems you can resolve this

Reply:
yes, the code in TestFwd is redundant